tsnet/tsnet_test.go

```diff
@@ -0,0 +1,8 @@
+package tsnet
+
+// ensure that the listener type always keeps the Server method, which is used
+// by some external applications to identify a tsnet.Listener from other
+// net.Listeners, as well as access the underlying Server.
+var _ interface {
+	Server() *Server
+} = new(listener)
```

Comment:
just make a regular TestFoo test?

Reply:
But if you'd rather check this at compile time, then put this in tsnet.go itself.

And write `= (*listener)(nil)`

Reply:
done